single-line-diagram-core/src/main/java/com/powsybl/sld/model/SwitchNode.java

```diff
@@ -50,5 +50,11 @@ public Node getOtherAdjNode(Node adj) {
     protected void writeJsonContent(JsonGenerator generator, boolean includeCoordinates) throws IOException {
         super.writeJsonContent(generator, includeCoordinates);
         generator.writeStringField("kind", kind.name());
+        if (this.getOrder().isPresent()) {
+            generator.writeNumberField("order", this.getOrder().get());
```

Comment:
You need to store the result of `getOrder` as nothing ensures the second call returns a non-empty optional. And therefore sonar detects it as a bug.

It would be even more readable to use `Optional::ifPresent`, but unfortunately you cannot because of the IOException